types/list.go

```diff
@@ -91,6 +91,16 @@ func (l ListType) Equal(o attr.Type) bool {
 	return l.ElemType.Equal(other.ElemType)
 }
 
+// ApplyTerraform5AttributePathStep applies the given AttributePathStep to the
+// list.
+func (l ListType) ApplyTerraform5AttributePathStep(step tftypes.AttributePathStep) (interface{}, error) {
+	if _, ok := step.(tftypes.ElementKeyInt); !ok {
+		return nil, fmt.Errorf("cannot apply step %T to ListType", step)
+	}
+
+	return l.ElemType, nil
+}
+
 // List represents a list of AttributeValues, all of the same type, indicated
 // by ElemType.
 type List struct {
```

Comment:
Do we need an `ApplyTerraform5AttributePathStep` on the `List` type, too?

Reply:
Why?

Reply:
Ugh okay I see. For some reason I thought `GetAttribute` would be traversing an `attr.Value`, but it's trying to _make_ an `attr.Value` at that point. So we need the `attr.Type` to be an AttributePathStepper (so we can get the relevant schema) and we need `tftypes.Value` to be an AttributePathStepper (so we can get the sub-`tftypes.Value` we're talking about) and then we can combine those two to make an `attr.Value`.

That makes sense. Sorry for the confusion here.